src/steam_network/protocol/protobuf_client.py

```diff
@@ -168,77 +321,211 @@ async def _get_obfuscated_private_ip(self) -> int:
+        #AccessToken = pollResponse.RefreshToken,
+        #ShouldSavePassword = True #err on side of caution in case this not being set causes them to ignore access token. then try false. 
+        """
+            var logon = new ClientMsgProtobuf<CMsgClientLogon>( EMsg.ClientLogon );
+
+            SteamID steamId = new SteamID( details.AccountID, details.AccountInstance, Client.Universe, EAccountType.Individual );
+
+            if ( details.LoginID.HasValue )
+            {
+                // TODO: Support IPv6 login ids?
+                logon.Body.obfuscated_private_ip = new CMsgIPAddress
+                {
+                    v4 = details.LoginID.Value
+                };
+            }
+            else
+            {
+                logon.Body.obfuscated_private_ip = NetHelpers.GetMsgIPAddress( this.Client.LocalIP! ).ObfuscatePrivateIP();
+            }
+
+            // Legacy field, Steam client still sets it
+            if ( logon.Body.obfuscated_private_ip.ShouldSerializev4() )
+            {
+                logon.Body.deprecated_obfustucated_private_ip = logon.Body.obfuscated_private_ip.v4;
+            }
+
+            logon.ProtoHeader.client_sessionid = 0;
+            logon.ProtoHeader.steamid = steamId.ConvertToUInt64();
+
+            logon.Body.account_name = details.Username; //Null
+            logon.Body.password = details.Password; //Null
+            logon.Body.should_remember_password = details.ShouldRememberPassword; //false
+
+            logon.Body.protocol_version = MsgClientLogon.CurrentProtocol; 
+            logon.Body.client_os_type = ( uint )details.ClientOSType; //
+            logon.Body.client_language = details.ClientLanguage;
+            logon.Body.cell_id = details.CellID ?? Client.Configuration.CellID //
+
+            logon.Body.steam2_ticket_request = details.RequestSteam2Ticket;
+
+            // we're now using the latest steamclient package version, this is required to get a proper sentry file for steam guard
+            logon.Body.client_package_version = 1771; // todo: determine if this is still required
+            logon.Body.supports_rate_limit_response = true;
+            logon.Body.machine_name = details.MachineName;
+            logon.Body.machine_id = HardwareUtils.GetMachineID( Client.Configuration.MachineInfoProvider );
+
+            // steam guard 
+            logon.Body.auth_code = details.AuthCode;
+            logon.Body.two_factor_code = details.TwoFactorCode;
+
+#pragma warning disable CS0618 // LoginKey is obsolete
+            logon.Body.login_key = details.LoginKey;
+#pragma warning restore CS0618 // LoginKey is obsolete
+
+            logon.Body.access_token = details.AccessToken;
+
+            logon.Body.sha_sentryfile = details.SentryFileHash;
+            logon.Body.eresult_sentryfile = ( int )( details.SentryFileHash != null ? EResult.OK : EResult.FileNotFound );
+
+
+            this.Client.Send( logon );
+        }
```

Comment:
that was lifted straight from steamkit. i was trying to use it to figure out why i couldn't get the login to work from the refresh token. I needed to set "should_save_password" to true iirc and wasn't. I actually found that by rebuilding DepotDownloader with a ton of debug print statements. In a commit on my dev brach i removed it because it's not relevant anymore. I'll remove it here too